notification.go

```diff
@@ -0,0 +1,97 @@
+package ts3
+
+import (
+	"strings"
+)
+
+type notifyEvent string
+
+// Notify event categorys
+const (
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `serveredited`.
+	ServerEvents notifyEvent = "server"
+
+	// ServerEvents registers the following events:
+	// `cliententerview`, `clientleftview`, `channeldescriptionchanged`, `channelpasswordchanged`
+	// `channelmoved`, `channeledited`, `channelcreated`, `channeldeleted`, `clientmoved`.
+	ChannelEvents notifyEvent = "channel"
+
+	// TextServerEvents registers the `textmessage` event with `targetmode = 3`.
+	TextServerEvents notifyEvent = "textserver"
+
+	// TextChannelEvents registers the `textmessage` event with `targetmode = 2`.
+	//
+	// Notifications are only received for messages that are sent in the channel that the client is in.
+	TextChannelEvents notifyEvent = "textchannel"
+
+	// TextPrivateEvents registers the `textmessage` event with `targetmode = 1`.
+	TextPrivateEvents notifyEvent = "textprivate"
+
+	// TokenUsedEvents registers the `tokenused` event.
+	TokenUsedEvents notifyEvent = "tokenused"
+)
+
+// Notification contains the information of a notify event.
+type Notification struct {
+	Type string
+	Data map[string]string
+}
+
+// Notifications returns a read-only channel that outputs received notifications.
+//
+// If you subscribe to server and channel events you will receive duplicate
+// `cliententerview` and `clientleftview` notifications.
+// Sending a private message from the client results in a `textmessage`
+// Notification even if the client didn't subscribe to any events.
+//
+// Notifications are not documented by TeamSpeak;
+// A complete but unofficial documentation in German can be found here:
+// http://yat.qa/ressourcen/server-query-notify/
+func (c *Client) Notifications() <-chan Notification {
+	return c.notify
+}
+
+// NotifyRegister subscribes for a specified category of events
```

Comment:
I'd go with simply Register similarly with Unregister below
`// Register registers for server event notifications.`